nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/NiFiClientUtil.java

```diff
@@ -1256,7 +1256,7 @@ public ProvenanceEntity waitForComplete(final ProvenanceEntity entity) throws In
         verificationRequest.setRequest(requestDto);
 
         VerifyConfigRequestEntity results = nifiClient.getReportingTasksClient().submitConfigVerificationRequest(verificationRequest);
-        while (!results.getRequest().isComplete()) {
+        while ((!results.getRequest().isComplete()) || (results.getRequest().getResults() == null)) {
```

Comment:
This covers a transitory state where the server reports `COMPLETED`, but before it had assembled the results.